### PR TITLE
Add native barometer support

### PR DESCRIPTION
**Pull request by CapnBry**
Adds receiver barometer support and telemetry generation.
* Added generic target `DIY_2400_RX_PWM_VARIO_via_X` / `Vario 2400.json` can be removed when hardware is announced 
* Support for SPL06 barometer

### devBaro
Adds a new ELRS device devBaro with a generic detection and working interface to allow other barometers to be used. The telemetry reports the Barometric Altitude as well as a Vario in cm/s via the `CRSF_FRAMETYPE_BARO_ALTITUDE` packet.

Note that due to the overhead of sending telemetry items in ExpressLRS, these two values are sent in the same telemetry item instead of splitting them between this one and `CRSF_FRAMETYPE_VARIO`. That would take twice the bandwidth while leaving empty payload space OTA since the payloads are just 2 bytes each, with 4 bytes of header/crc. This requires EdgeTX 2.8.0 (currently in RC) to have them both display, as this "extended baro alt" was a recent feature.

When a vario is detected, the update rate of VBAT is lowered from 2Hz to 1Hz to reserve more bandwidth for the Vario/Baro sensor.

#### Update Rate (SPL06)
The current internal sampling rate of the SPL06 barometer runs in 70ms, with 32x oversampling on the pressure (0.9PaRMS), and 8x on the temperature. The altitude is reported without smoothing and has a RMS precision of about 0.8m, but the vario is smoothed `(old * 3 + new) / 4` due to the high update rate making projecting a 70ms sample to cm/s have an unusable amount of overshoot.

While the internal update rate is fixed and runs at a steady period, the actual update to the handset is dependent on telemetry bandwidth. Measured update rates at various rates and ratios (333Hz and 100Hz are FullRes). I believe this can be improved in OTA4 with more optimization, but 10Hz update rate is still achievable 
| Packet Rate | TLM Ratio | Update Interval (ms) | Avg Interval (ms) |
|---|---|---|---|
| 333Hz | 1:128  | 768-1537 | 1470 |
| 333Hz | 1:64 | 384-769 | 620 |
| 333Hz | 1:32 | 192-385 | 240 |
| 333Hz | 1:16 | 96-192 | 125 |
| 333Hz | 1:8 | 72-96 |  84 |
| 333Hz | 1:4 | 83-85 | 84 |
| 333Hz | 1:2 | 83-85 | 84 |
| 150Hz | 1:32 | 854-2133 | 1550 |
| 150Hz | 1:16 | 320-747 | 500 |
| 150Hz | 1:8 | 160-267 | 183 |
| 150Hz | 1:4 | 80-133 | 92 |
| 150Hz | 1:2 | 80-94 | 83 |
| 100Hz | 1:32 | 640-1280 | 1000 |
| 100Hz | 1:16 | 320-640 | 460 |
| 100Hz | 1:8 | 160-320 | 205 |
| 100Hz | 1:4 | 80-160 | 108 |
| 100Hz | 1:2 | 80-100 | 84 |

Note that this (and all ESP32 RX) run very hot, so 1:2 ratio is not recommended.

### VBAT calculation
This receiver needed to have a negative `VBAT_OFFSET` defined (_adding_ a value to the ADC reading), which was not accounted for in the reporting. The updated code reports any value where the unmodified ADC reading was 0 as 0V. 
image

The alternative would be to calculate the value anyway, but clamp it at 0V minimum, but the linearity of the ESP32 ADC at 11dB attenuation factor is bad so values in this range would likely be incorrect anyway. We likely need a better system than this simple linear two-point calibration at some point, but this does a fairly ok job with this receiver. The fit could be improved if it didn't try to support up to 35V (rolls eyes)
image

### Extended Changes
* VBAT telemetry is reduced after a baro is detected, an interface function was added to enable this mode
* Disable UART0 pins in devServoOutput on ESP32 when compiled with DEBUG_LOG (ESP8266 already did this too)
* Disable UART0 pins in devServoOutput if `DEBUG_RCVR_LINKSTATS` is defined, as it needs the uart to report as well

### CODE REVIEW

File: src/html/hardware.html [hunk 1]
```diff
@@ -271,7 +271,11 @@ <h1><b>ExpressLRS</b></h1>
 					<tr><td></td><td>SPI MOSI pin<img class="icon-output"/></td><td><input size='3' id='vtx_mosi' name='vtx_mosi' type='text'/></td><td>MOSI pin on RTC6705 VTx (leave undefined if sharing Radio SPI bus)</td></tr>
 					<tr><td></td><td>25mW VPD interpolation values</td><td><input size='40' id='vtx_amp_vpd_25mW' name='vtx_amp_vpd_25mW' type='text' class='array'/></td><td>4 values for 5650, 5750, 5850, 5950 frequencies at 25mW</td></tr>
 					<tr><td></td><td>100mW VPD interpolation values</td><td><input size='40' id='vtx_amp_vpd_100mW' name='vtx_amp_vpd_100mW' type='text' class='array'/></td><td>4 values for 5650, 5750, 5850, 5950 frequencies at 100mW</td></tr>
-@@end
+
+					<tr><td colspan='2'><b>I2C</td></tr>
+					<tr><td></td><td>SCL pin<img class="icon-output"/></td><td><input size='3' id='i2c_scl' name='i2c_scl' type='text'/></td><td>I2C clock pin used to communicate with I2C devices</td></tr>
+					<tr><td></td><td>SDA pin<img class="icon-input"/><img class="icon-output"/></td><td><input size='3' id='i2c_sda' name='i2c_sda' type='text'/></td><td>I2C data pin used to communicate with I2C devices</td></tr>
+	@@end
```
Comment (pkendall64) [inline src/html/hardware.html:278]: Move to beginning of line like all the other template tags

**Review by pkendall64 — CHANGES_REQUESTED** [review]

File: src/html/hardware.html (hunk 1 above)
Comment (CapnBry) [reply src/html/hardware.html]: Oops stupid whitespace!

**Review by pkendall64 — APPROVED** [review]

File: src/hardware/targets.json [hunk 2]
```diff
@@ -487,6 +487,14 @@
                 "firmware": "Unified_ESP8285_2400_RX",
                 "prior_target_name": "DIY_2400_RX_PWMPEX"
             },
+            "pwm-vario": {
+                "product_name": "Generic ESP32 PWM Vario 2.4Ghz RX",
+                "lua_name": "Vario 2400RX",
+                "layout_file": "Vario 2400.json",
+                "upload_methods": ["uart", "wifi", "betaflight"],
```
Comment (jurgelenas) [inline src/hardware/targets.json:494]: Is betaflight passthrough really relevant for this one?

Comment (CapnBry) [reply src/hardware/targets.json:494]: It isn't strictly, but Betaflight Passthrough is just "via_UART except for an up-and-running receiver". I left it in for when the Outputs can be configured as CRSF TX/RX eventually, but I can remove it for now and add it back when that feature is available.

**Review by jurgelenas — APPROVED** [review]

**Review by FOG-Yamato — APPROVED** [review]

**Pull request merged**